mac/TestInput/InDesign TestInput Instructions.md

```diff
@@ -0,0 +1,158 @@
+compatibility with input methods that use Apple's standard approach using InputMethodKit.
+
+This is of particular concern to us because we have developed [Keyman](keyman.com), a very popular and
+widely used input method that supports custom keyboards in hundreds of languages,
+including many major languages whose complex scripts are best served by the powerful
+options in Keyman as well as minority languages whose only viable option for text input
+might be Keyman.
+
+We have prepared a small sample app that easily illustrates these incompatibilities.
+Like Keyman, the sample app uses the _third_ approach for receiving and processing text
+events as described [here](https://developer.apple.com/documentation/inputmethodkit/imkserverinput?language=objc).
+
+Note that our testing was done with InDesign CC 2018 version 13.1.
+
+For each of the four problems, follow the testing steps described below to see the problem.
+
+# Installation and Setup of TestInput sample app
+
+1. Unzip the **InDesign IM Incompatibilities Test.zip** file
+2. Copy TestInput.app into **~/Library/Input Methods**
+3. Open **System Preferences**.
+4. Open **Keyboard**.
+5. Click the **Input Sources** tab.
+6. Click **+** to add an input source.
+7. In the list of languages, choose **Multiple Languages**.
+8. In the righthand pane, you should see **TestInput** (its icon is a **9**). Select it. 
+9. Click **Add**.
+10. Ensure that the checkbox **Show input menu in menu bar** is selected.
+11. Now the menu bar should include an icon for the input menu. If U.S. English is your default system input method, the icon will be an American flag.
+12. Click the input menu icon to drop down that menu and confirm that **TestInput** appears as a choice. Select it.
+13. The icon for the input menu should change to a **9**. Drop the menu again to confirm that there is now a new section in the menu called **TestInput**, which has three items.
+14. Now you are ready to go on the next section to perform the steps to see the problems.
+
+Note: _You can leave **TestInput** selected as the input source during the entire process, but if you need to type anything (e.g., to take notes) during the process, you will probably find it beneficial to switch back to your default system keyboard to do so, as **TestInput** does some funny things in response to several keys._
+
+# Steps to Illustrate Compatibility Problems in InDesign
+
+## 1. Actions associated with Submenus on the input menu do not fire
+
+Steps to reproduce:
+
+1. Activate InDesign.
+2. For this test, it doesn't seem to matter what the context is in InDesign. We have tested on a blank page, in a text box, the **Get Started** welcome screen, and with the insertion point in a text box on the formatting toolbar.
+3. Ensure that your system volume is turned up load enough for you to be able to hear system sounds.
+4. _Optionally_, start Console and filter to show all messages from the **TestUInput** process.
```

Comment:
`TestUInput` or `TestInput`? Here and in 5 other places.

Reply:
Good catch